wrapanapi/systems/container/rhopenshift.py

```diff
@@ -604,9 +603,9 @@ def rename_structure(self, struct):
             struct: data to process and rename
         Return: updated data
         """
-        if not isinstance(struct, six.string_types) and isinstance(struct, Iterable):
+        if not isinstance(struct, str) and isinstance(struct, Iterable):
             if isinstance(struct, dict):
-                for key in struct.keys():
+                for key in list(struct.keys()):
```

Comment:
This should be redundant in py3, but you're using `str` above like its written for py3 only, I'm confused?

Reply:
so, keys are some kind of iterator now. I have to make static. this change just makes this list fixed.

Reply:
Some outside discussion cleared this up - the reason for this isn't iteration over the keys themselves, but the fact that `struct` is being modified within the `for` loop, and thus its size changes during iteration.  Explicitly wrapping the `keys()` iterator causes the `for` loop to traverse the list object instead, which will not change size when `struct` is modified.